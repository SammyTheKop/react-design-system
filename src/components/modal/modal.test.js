// react
import React from "react";

// component
import Modal from "./modal";

// jest renderer
import renderer from "react-test-renderer";

// testing library react
import "@testing-library/jest-dom/extend-expect";

// Case 1: Button Renders properly or not
it("Case 1: Button Component Renders Correctly", () => {
  const modalComponentTree = renderer.create(<Modal>Test</Modal>).toJSON();
  expect(modalComponentTree).toMatchSnapshot();
});
