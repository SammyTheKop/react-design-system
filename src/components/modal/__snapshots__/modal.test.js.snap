// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Case 1: Button Component Renders Correctly 1`] = `
Array [
  <div
    className="modal"
    data-testid="modal-container"
  >
    <div
      className="modal-content medium"
    >
      <span
        className="close-icon"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        Ã—
      </span>
      Test
    </div>
  </div>,
  <style
    jsx={true}
  >
    
          /* Styles for the Modal Component */
          .modal {
            display: none; /* Hidden by default */
            position: fixed; /* Stay in place */
            z-index: 999; /* Sit on top */
            left: 0;
            top: 0;
            width: 100%; /* Full width */
            height: 100%; /* Full height */
            overflow: auto; /* Enable scroll if needed */
            background-color: rgba(0, 0, 0, 0.4); /* Black w/ opacity */
          }

          .modal .modal-content {
            background-color: #fefefe;
            position: relative;
            border-radius: 3px;
            margin: auto;
            overflow: hidden;
          }

          .modal-content.large {
            width: 90%;
            height: 90%;
          }

          .modal-content.medium {
            width: 75%;
            height: 75%;
          }

          .modal-content.small {
            width: 50%;
            width: 50%;
          }

          .modal .modal-content.fadein {
            animation-name: fadein;
            animation-duration: 0.4s;
            display: block;
          }

          .modal .modal-header {
            padding: 15px;
            text-align: left;
            border-bottom: 1px solid #ececec;
          }

          .modal .modal-body {
            text-align: left;
            min-height: 100px;
          }

          .modal .modal-footer {
            padding: 15px;
            text-align: left;
            border-top: 1px solid #e5e5e5;
            background: #ececec;
            border-radius: 0 0 8px 8px;
          }

          .close-icon {
            position: absolute;
            right: 15px;
            top: 5px;
            cursor: pointer;
            font-size: 30px;
          }

          .modal.show {
            display: flex;
          }

          /* Add Animation */
          @keyframes fadein {
            from {
              opacity: 0;
            }
            to {
              opacity: 1;
            }
          }
        
  </style>,
]
`;
